File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.py
N, K = map(int, input().split())

arr = list(range(1, N + 1))
idx = 0
result = []
while len(arr) > 0:
  idx = (idx + K - 1) % len(arr)
  result.append(arr.pop(idx))

print("<" + ", ".join(map(str, result)) + ">")

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.js
const fs = require("fs");

const input = fs.readFileSync(0, "utf-8").trim().split(" ");

const N = parseInt(input[0]);
const K = parseInt(input[1]);

let idx = 0;
const arr = Array.from({ length: N }, (_, i) => i + 1);
const result = [];

for (let i = 1; i <= N; i++) {
  idx = (idx + K - 1) % arr.length;
  let removeEl = arr.splice(idx, 1);
  result.push(removeEl[0]);
}

console.log(`<${result.join(", ")}>`);
